### PR TITLE
Split db from util

### PR DESCRIPTION
**Pull request by zeroqn**
cita-bft, cita-forever, cita-jsonrpc don't need to build rocksdb.

### CODE REVIEW

File: util/src/lib.rs [hunk 1]
```diff
@@ -1,5 +1,5 @@
 // CITA
-// Copyright 2016-2017 Cryptape Technologies LLC.
+// Copyright 2016-2017 Cryptape Tecrhnologies LLC.
```
Comment (u2) [inline util/src/lib.rs]: ?

**Review by u2 — CHANGES_REQUESTED** [review]

File: util/src/lib.rs (hunk 1 above)
Comment (zeroqn) [reply util/src/lib.rs]: Fix. May be my shortcut problem.

**Comment by u2** [comment]
I consider we could move `kvdb` in a separate crate. What about removing the `avl` or moving to another repo firstly, because it is written base on the `trie`, it's not well designed.

**Review by u2 — APPROVED** [review]

**Comment by zeroqn** [comment]
That's my first try, move kvdb. But later i found out that all of dbs are based on hashdb.rs, i haven't figured
out how to deal with this. Move this file to kvdb too?  If we don't plan to revisit current avl code in the future then it's ok to remove it.

**Comment by u2** [comment]
We have not delivery the avl to CITA until now and have no plan about it recently.

**Pull request merged**